### ISSUE TITLE
Measurement: Allow to project measures

### ISSUE THREAD

**Issue by pierreporte** [FreeCAD/FreeCAD#13741]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

The new measurement feature gives the value of the measure and nothing else. It is sometimes useful to determine the 3 components of a measure in the base coordinate system. It may be desirable to determine said components in an arbitrary LCS, or the projected value on an user-selected plane or line.

Commercial CAD programs often provide the Δx, Δy, and Δz values for a distance, as you can see on the Solidworks screenshot below, but FreeCAD could be more complete

![image](https://github.com/FreeCAD/FreeCAD/assets/2029853/d58a3b18-6505-4c9a-bf73-d14c3895ba4e)

### Full version info

```shell
OS: Ubuntu 23.10 (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37063 (Git) AppImage
Build type: Release
Branch: main
Hash: 680792030fd664c94829e2c1b78b21788e8d9879
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: French/France (fr_FR)
Installed mods: 
  * BillOfMaterials-WB 0.0.7
  * CosmeticThread3D
  * OpenTheme 2024.4.20
  * Curves 0.6.35
  * CurvedShapes 1.0.9
  * sheetmetal 0.4.13
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
Is it using the global coordinate system or the one from the part/body or with a picking direction?

**Comment by pierreporte** [comment]
All of that, depending on user choice.

**Comment by FEA-eng** [comment]
@hlorus Is there a chance that this will make it to the upcoming release (so essentially before the feature freeze on June 3rd)? It's a very important feature that was available in the old measurement tools and users are already asking for it: https://forum.freecad.org/viewtopic.php?p=759615#p759539

**Comment by adrianinsaval** [comment]
This could be considered a regression so maybe we can make an exception and still accept something like this after that, it would certainly have my support. But ideally it should happen before

**Comment by FEA-eng** [comment]
@hlorus Maybe (at least for now) it would be possible to add projected (XYZ) measurement values without displaying them in the same way as regular dimensions (with lines and arrows). So just the additional 3 values would be shown either within the label like here:

![image](https://github.com/FreeCAD/FreeCAD/assets/59876896/62c650f8-bf8c-420d-b36f-91be648dd1fb)

or in the report view:

![components](https://github.com/FreeCAD/FreeCAD/assets/59876896/c8411473-2e36-4913-b1ad-5e0a8de283ae)

That should be sufficient to get rid of this important regression and then it could be potentially improved in the future.

**Comment by maxwxyz** [comment]
@pierreporte only the delta in the document coordinate system XYZ is possible now. There is no way of projecting measures by choosing local coordinate systems or references. Should this issue be reopened?

**Comment by FEA-eng** [comment]
> @pierreporte only the delta in the document coordinate system XYZ is possible now. There is no way of projecting measures by choosing local coordinate systems or references. Should this issue be reopened?

It was like that with the old measurement tools too so might be better to make a new FR.

**Comment by pierreporte** [comment]
I’ve created a follow-up issue: #16991.
